sentence_transformers/sampler.py

```diff
@@ -53,32 +70,28 @@ def __init__(
         if self.batch_size % 2 == 1:
             raise ValueError("The batch size for `GroupByLabelBatchSampler` must be divisible by 2.")
 
-        for column_name in valid_label_columns or []:
-            if column_name in dataset.column_names:
-                labels = dataset["label"]
-                break
-        else:
-            raise ValueError(f"None of the valid_label_columns {valid_label_columns} are in the dataset.")
-
-        del dataset
+        labels = self._determine_labels_to_use(dataset, valid_label_columns)
         groups = defaultdict(list)
         for sample_idx, label in enumerate(labels):
             groups[label].append(sample_idx)
+        self.groups = groups
 
```

Comment:
```suggestion
        self.groups = {
            label: sample_indices[:num_samples]
            for label, sample_indices in groups.items()
            if (num_samples := len(sample_indices) // 2 * 2)
        }
```

See https://github.com/UKPLab/sentence-transformers/issues/2782#issuecomment-2196579410

Ideally, this should be tested as well. e.g.:
```python
@pytest.fixture
def dummy_uneven_dataset():
    """
    Dummy dataset for testing purposes. The dataset looks as follows:
    {
        "data": ["a"] * 51,
        "label": [0] * 17 + [1] * 17 + [2] * 17,
    }
    """
    data = {"data": ["a"] * 51, "label": [0] * 17 + [1] * 17 + [2] * 17}
    return Dataset.from_dict(data)
```
with a `GroupByLabelBatchSampler` with batch size of 16. Each batch should still always have 2+ texts for each label.